torch/onnx/diagnostic/rules.yaml

```diff
@@ -0,0 +1,81 @@
+# Rules are based on sarif ReportingDescriptor format.
+# TODO: Define additional format requirements on top of sarif for our usage.
+#       For example: pre defined keys for message_strings for logging on different levels.
+# TODO: Based on above, create helper script to generate new rules.
+# TODO: Separate rules into individual files.
+
+- id: RL0001
+  name: ONNXShapeInferenceIsMissingForNode
```

Comment:
Any consideration for casing style from sarif? Have we considered snake_case? (More pythonic, aligns with mypy etc.)

Reply:
sarif format itself in json is camel cased, it is converted to snake case when generating the python object model. Yet this is somewhat in parallel to the format for actual name value, which is what I think your comment is referring to.

I like the dash style you are proposing, I'll push a change to see how it looks.